R/uvhydrograph-fnc.R

```diff
@@ -1,133 +1,141 @@
 
 #'@export
 uvhydrographPlot <- function(data){
-  uv_pts <- getUvHydro(data, "primarySeries" )
-  layout_uvhydro()
-  
-  primary_lbl <- getUvLabel(data, "primarySeries")
-  
-  add_uvhydro_axes(getUvhLims(uv_pts), ylab = primary_lbl, ylog = TRUE)
-  
-  dv_pts <- getUvHydro(data, "derivedSeriesMean" )
-  dv_pts$x = dv_pts$x + 86400/2 # manual shift for now...
-  
-  # discharge measurements and errors
-  if(isSeriesOfType(data, "primarySeries", "Discharge")) {
-    add_q_measurements(data)
-  }
-  
-  add_computed_uv(uv_pts)
-  add_approved_dv(dv_pts)
-  
-  secondary_lbl <- getUvLabel(data, "secondarySeries")
-  tertiary_lbl <- getUvLabel(data, "effectiveShifts")
-  
-  uv2_pts <- getUvHydro(data, "secondarySeries")
-  shift_pts <- getUvHydro(data, "effectiveShifts")
-  
-  secondary_lims <- getUvhLims(uv2_pts)
-  add_uvhydro_axes(secondary_lims, ylab = secondary_lbl, ylog = FALSE)
+  #for pagination by month, get all of the month strings for the primary series
+  all_primary_pts <- getUvHydro(data, "primarySeries" )
+  months <- unique(all_primary_pts$month, incomparables = FALSE)
   
-  add_computed_uv(uv2_pts)
-  shifted_pts <- uv2_pts
-  shifted_pts[['y']] <- shifted_pts[['y']] + shift_pts[['y']]
-  add_edited_uv(shifted_pts)
+  layout_uvhydro()
   
-  # gageHeight
-  if(isSeriesOfType(data, "secondarySeries", "Gage height")) {
-    add_stage_measurements(data)
-    
-    #add effective shift axis, timeseries, and shift measurements
-    # NOTE: this is a third plot, so this has to come at the end of this method.
-    #third axis
-    measuredShifts <- getFieldVisitErrorBarsShifts(data)
-    add_third_axes(secondary_lims = secondary_lims, tertiary_pts = shift_pts, tertiary_lbl = tertiary_lbl, measured_shift_pts = measuredShifts)
-    add_shift_measurements(measuredShifts)
+  for (month in months){  
+    #legend vector, needs to be dynamically built up with plots
+    primary_legend <- getNewLegendFrame()
+    addToPrimaryLegend <- function(newText, newSymbol, newColor, newLine) { 
+      primary_legend <<- rbind(primary_legend, data.frame(text = newText, symbol = newSymbol, color = newColor, line = newLine, stringsAsFactors = FALSE))
```

Comment:
Yeah, this is a "javascript pattern", trying to pass in a "closure" to the functions so that they can call a function which updates the variable in the environment above. Given what @jiwalker-usgs said above, and some discussion with @ldecicco-USGS we think this is actually not global.